In [1]:
import pandas as pd

df_train = pd.read_csv('train.csv')
df_train.shape

(4209, 378)

In [2]:
pd.options.display.max_columns = 999
df_train.head()

ID       y  X0 X1  X2 X3 X4 X5 X6 X8  X10  X11  X12  X13  X14  X15  X16  \
0   0  130.81   k  v  at  a  d  u  j  o    0    0    0    1    0    0    0   
1   6   88.53   k  t  av  e  d  y  l  o    0    0    0    0    0    0    0   
2   7   76.26  az  w   n  c  d  x  j  x    0    0    0    0    0    0    0   
3   9   80.62  az  t   n  f  d  x  l  e    0    0    0    0    0    0    0   
4  13   78.02  az  v   n  f  d  h  d  n    0    0    0    0    0    0    0   

   X17  X18  X19  X20  X21  X22  X23  X24  X26  X27  X28  X29  X30  X31  X32  \
0    0    1    0    0    1    0    0    0    0    0    0    0    0    1    0   
1    0    1    0    0    0    0    0    0    0    1    0    0    0    1    0   
2    1    0    0    0    0    0    0    0    0    1    1    1    0    1    0   
3    0    0    0    0    0    0    0    0    0    1    1    1    0    1    0   
4    0    0    0    0    0    0    0    0    0    1    1    1    0    1    0   

   X33  X34  X35  X36  X37  X38  X39  X40  X41  X42  X43  X44  X45  X46  X47  \
0    0    0    1    0    1    0    0    0    0    0    0    0    0    1    0   
1    0    0    1    0    1    0    0    0    0    0    0    0    0    0    0   
2    0    0    1    0    1    0    0    0    0    0    1    0    0    1    0   
3    0    0    1    0    1    0    0    0    0    0    1    0    0    1    0   
4    0    0    1    0    1    0    0    0    0    0    1    0    0    1    0   

   X48  X49  X50  X51  X52  X53  X54  X55  X56  X57  X58  X59  X60  X61  X62  \
0    0    0    0    0    0    0    0    0    0    0    1    0    0    0    0   
1    0    0    0    1    0    0    0    0    0    0    0    0    0    1    0   
2    0    0    0    1    0    0    1    0    0    0    1    0    0    1    0   
3    0    0    0    0    0    0    1    0    0    0    0    0    0    1    0   
4    0    0    0    1    0    0    1    0    0    0    0    0    0    1    0   

   X63  X64  X65  X66  X67  X68  X69  X70  X71  X73  X74  X75  X76  X77  X78  \
0    0    0    0    0    0    1    0    1    0    0    1    0    0    0    0   
1    0    0    0    0    0    0    0    1    0    0    1    0    0    0    0   
2    0    0    0    0    0    1    0    1    1    0    1    1    1    0    0   
3    0    0    0    0    0    0    0    1    1    0    1    0    1    0    0   
4    0    0    0    0    0    0    0    1    0    0    1    0    1    0    0   

   X79  X80  X81  X82  X83  X84  X85  X86  X87  X88  X89  X90  X91  X92  X93  \
0    0    0    0    0    0    0    1    0    0    0    0    0    0    0    0   
1    0    1    0    0    0    0    1    0    0    0    0    0    0    0    0   
2    0    1    0    0    0    1    1    0    0    0    0    0    0    0    0   
3    0    1    0    0    0    1    0    0    0    0    0    0    0    0    0   
4    0    1    0    0    0    0    0    0    0    0    0    0    0    0    0   

   X94  X95  X96  X97  X98  X99  X100  X101  X102  X103  X104  X105  X106  \
0    0    0    0    0    0    0     0     0     0     0     0     0     0   
1    0    0    1    0    1    0     1     1     0     0     0     0     0   
2    0    0    1    0    1    0     0     1     0     0     0     0     0   
3    0    0    1    0    1    0     0     1     0     0     0     0     0   
4    0    0    1    0    1    0     0     1     0     0     0     0     0   

   X107  X108  X109  X110  X111  X112  X113  X114  X115  X116  X117  X118  \
0     0     0     0     0     1     0     0     1     0     1     0     1   
1     0     0     0     0     1     0     0     0     0     0     0     1   
2     0     0     0     0     1     0     0     0     0     0     0     0   
3     0     1     0     0     1     0     0     1     0     0     0     0   
4     0     1     0     0     1     0     0     1     0     0     0     0   

   X119  X120  X122  X123  X124  X125  X126  X127  X128  X129  X130  X131  \
0     1     1     0     0     0     0     0     0     1     0     0     1   
1     1     1     0     0     0     0     0     1

In [3]:
df_train_y = df_train['y']
df_train_x = df_train.drop(columns = ['y'])

df_train_x.dtypes.value_counts()

int64     369
object      8
dtype: int64

As observed, 8 categoritcal features from X0 to X8, with remaining 369 features are of integer type

In [4]:
df_train_x.dtypes.to_frame('Type').T

ID      X0      X1      X2      X3      X4      X5      X6      X8  \
Type  int64  object  object  object  object  object  object  object  object   

        X10    X11    X12    X13    X14    X15    X16    X17    X18    X19  \
Type  int64  int64  int64  int64  int64  int64  int64  int64  int64  int64   

        X20    X21    X22    X23    X24    X26    X27    X28    X29    X30  \
Type  int64  int64  int64  int64  int64  int64  int64  int64  int64  int64   

        X31    X32    X33    X34    X35    X36    X37    X38    X39    X40  \
Type  int64  int64  int64  int64  int64  int64  int64  int64  int64  int64   

        X41    X42    X43    X44    X45    X46    X47    X48    X49    X50  \
Type  int64  int64  int64  int64  int64  int64  int64  int64  int64  int64   

        X51    X52    X53    X54    X55    X56    X57    X58    X59    X60  \
Type  int64  int64  int64  int64  int64  int64  int64  int64  int64  int64   

        X61    X62    X63    X64    X65    X66    X67    X68    X69    X70  \
Type  int64  int64  int64  int64  int64  int64  int64  int64  int64  int64   

        X71    X73    X74    X75    X76    X77    X78    X79    X80    X81  \
Type  int64  int64  int64  int64  int64  int64  int64  int64  int64  int64   

        X82    X83    X84    X85    X86    X87    X88    X89    X90    X91  \
Type  int64  int64  int64  int64  int64  int64  int64  int64  int64  int64   

        X92    X93    X94    X95    X96    X97    X98    X99   X100   X101  \
Type  int64  int64  int64  int64  int64  int64  int64  int64  int64  int64   

       X102   X103   X104   X105   X106   X107   X108   X109   X110   X111  \
Type  int64  int64  int64  int64  int64  int64  int64  int64  int64  int64   

       X112   X113   X114   X115   X116   X117   X118   X119   X120   X122  \
Type  int64  int64  int64  int64  int64  int64  int64  int64  int64  int64   

       X123   X124   X125   X126   X127   X128   X129   X130   X131   X132  \
Type  int64  int64  int64  int64  int64  int64  int64  int64  int64  int64   

       X133   X134   X135   X136   X137   X138   X139   X140   X141   X142  \
Type  int64  int64  int64  int64  int64  int64  int64  int64  int64  int64   

       X143   X144   X145   X146   X147   X148   X150   X151   X152   X153  \
Type  int64  int64  int64  int64  int64  int64  int64  int64  int64  int64   

       X154   X155   X156   X157   X158   X159   X160   X161   X162   X163  \
Type  int64  int64  int64  int64  int64  int64  int64  int64  int64  int64   

       X164   X165   X166   X167   X168   X169   X170   X171   X172   X173  \
Type  int64  int64  int64  int64  int64  int64  int64  int64  int64  int64   

       X174   X175   X176   X177   X178   X179   X180   X181   X182   X183  \
Type  int64  int64  int64  int64  int64  int64  int64  int64  int64  int64   

       X184   X185   X186   X187   X189   X190   X191   X192   X194   X195  \
Type  int64  int64  int64  int64  int64  int64  int64  int64  int64  int64   

       X196   X197   X198   X199   X200   X201   X202   X203   X204   X205  \
Type  int64  int64  int64  int64  int64  int64  int64  int64  int64  int64   

       X206   X207   X208   X209   X210   X211   X212   X213   X214   X215  \
Type  int64  int64  int64  int64  int64  int64  int64  int64  int64  int64   

       X216   X217   X218   X219   X220   X221   X222   X223   X224   X225  \
Type  int64  int64  int64  int64  int64  int64  int64  int64  int64  int64   

       X226   X227   X228   X229   X230   X231   X232   X233   X234   X235  \
Type  int64  int64  int64  int64  int64  int64  int64  int64  int64  int64   

       X236   X237   X238   X239   X240   X241   X242   X243   X244   X245  \
Type  int64  int64  int64  int64  int64  int64  int64  int64  int64  int64   

       X246   X247   X248   X249   X250   X251   X252   X253   X254   X255  \
Type  int64  int64  int64  int64  int64  int64  int64  int64  int64  int64   

       X256   X257   X258   X259   X260   X261   X262   X263   X264   X265  \
Type

In [5]:
df_train_x.describe()

ID          X10     X11          X12          X13  \
count  4209.000000  4209.000000  4209.0  4209.000000  4209.000000   
mean   4205.960798     0.013305     0.0     0.075077     0.057971   
std    2437.608688     0.114590     0.0     0.263547     0.233716   
min       0.000000     0.000000     0.0     0.000000     0.000000   
25%    2095.000000     0.000000     0.0     0.000000     0.000000   
50%    4220.000000     0.000000     0.0     0.000000     0.000000   
75%    6314.000000     0.000000     0.0     0.000000     0.000000   
max    8417.000000     1.000000     0.0     1.000000     1.000000   

               X14          X15          X16          X17          X18  \
count  4209.000000  4209.000000  4209.000000  4209.000000  4209.000000   
mean      0.428130     0.000475     0.002613     0.007603     0.007840   
std       0.494867     0.021796     0.051061     0.086872     0.088208   
min       0.000000     0.000000     0.000000     0.000000     0.000000   
25%       0.000000     0.000000     0.000000     0.000000     0.000000   
50%       0.000000     0.000000     0.000000     0.000000     0.000000   
75%       1.000000     0.000000     0.000000     0.000000     0.000000   
max       1.000000     1.000000     1.000000     1.000000     1.000000   

               X19          X20          X21          X22          X23  \
count  4209.000000  4209.000000  4209.000000  4209.000000  4209.000000   
mean      0.099549     0.142789     0.002613     0.086957     0.020670   
std       0.299433     0.349899     0.051061     0.281805     0.142294   
min       0.000000     0.000000     0.000000     0.000000     0.000000   
25%       0.000000     0.000000     0.000000     0.000000     0.000000   
50%       0.000000     0.000000     0.000000     0.000000     0.000000   
75%       0.000000     0.000000     0.000000     0.000000     0.000000   
max       1.000000     1.000000     1.000000     1.000000     1.000000   

               X24          X26          X27          X28          X29  \
count  4209.000000  4209.000000  4209.000000  4209.000000  4209.000000   
mean      0.001901     0.004989     0.682585     0.032549     0.043003   
std       0.043561     0.070467     0.465526     0.177475     0.202888   
min       0.000000     0.000000     0.000000     0.000000     0.000000   
25%       0.000000     0.000000     0.000000     0.000000     0.000000   
50%       0.000000     0.000000     1.000000     0.000000     0.000000   
75%       0.000000     0.000000     1.000000     0.000000     0.000000   
max       1.000000     1.000000     1.000000     1.000000     1.000000   

               X30          X31          X32          X33          X34  \
count  4209.000000  4209.000000  4209.000000  4209.000000  4209.000000   
mean      0.004514     0.232359     0.011167     0.000238     0.005464   
std       0.067043     0.422387     0.105093     0.015414     0.073729   
min       0.000000     0.000000     0.000000     0.000000     0.000000   
25%       0.000000     0.000000     0.000000     0.000000     0.000000   
50%       0.000000     0.000000     0.000000     0.000000     0.000000   
75%       0.000000     0.000000     0.000000     0.000000     0.000000   
max       1.000000     1.000000     1.000000     1.000000     1.000000   

               X35          X36          X37          X38          X39  \
count  4209.000000  4209.000000  4209.000000  4209.000000  4209.000000   
mean      0.232359     0.004514     0.232359     0.033262     0.000238   
std       0.422387     0.067043     0.422387     0.179341     0.015414   
min       0.000000     0.000000     0.000000     0.000000     0.000000   
25%       0.000000     0.000000     0.000000     0.000000     0.000000   
50%       0.000000     0.000000     0.000000     0.000000     0.000000   
75%       0.000000     0.000000     0.000000     0.000000     0.000000   
max       1.000000     1.000000     1.000000     1.000000     1.000000   

               X40          X41          X42          X43  

In [17]:
df_description = df_train_x.describe().T.reset_index()
df_description.groupby(['min', 'max']).count()

index  count  mean  std  25%  50%  75%
min max                                           
0.0 0.0        12     12    12   12   12   12   12
    1.0       356    356   356  356  356  356  356
    8417.0      1      1     1    1    1    1    1

Out of the 369 features of integer type, 12 are all 0, 1 is the 'ID' and the remaining 356 are binary with min value 0 and max value 1

In [34]:
list_empty_features = df_description[df_description['min']==df_description['max']]['index'].to_list()
print("Features with single value: ", list_empty_features)

Features with single value:  ['X11', 'X93', 'X107', 'X233', 'X235', 'X268', 'X289', 'X290', 'X293', 'X297', 'X330', 'X347']
